<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>882</width>
    <height>683</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset resource="images.qrc">
    <normaloff>:/images/icon.png</normaloff>:/images/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4" rowstretch="1,2" columnstretch="1,1">
    <item row="1" column="1">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="title">
       <string>其他功能</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>数据库</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QPushButton" name="btn_CreateSQL">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>创建数据表</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_SortSQL">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>排序</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_PrintSQL">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>打印/删除</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_ClearSQL">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>清空</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit_SQL"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>写入文件</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="btn_WriteFile">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>创建和写入</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_OpenFile">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>打开文件夹</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_MyWidget">
      <property name="title">
       <string>MyWidget自定义绘制</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="MyWidget" name="myWidget" native="true"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>工具栏自定义输出</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPlainTextEdit" name="plainTextEdit_toolBar"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="table_window">
       <attribute name="title">
        <string>表格</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QTableWidget" name="tableWidget"/>
        </item>
        <item>
         <widget class="QPushButton" name="btn_Reverse">
          <property name="text">
           <string>倒序显示(按学号)</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tree_window">
       <attribute name="title">
        <string>树形图</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QTreeView" name="treeView">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="btn_ChangeNum">
            <property name="text">
             <string>小于25的数变为0</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_DeleteRow">
            <property name="text">
             <string>过滤单数行号</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>882</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Output1"/>
   <addaction name="action_Output2"/>
   <addaction name="action_Output3"/>
   <addaction name="action_Output4"/>
   <addaction name="action_Output5"/>
  </widget>
  <action name="action_Output1">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/output1.svg</normaloff>:/images/output1.svg</iconset>
   </property>
   <property name="text">
    <string>输出1</string>
   </property>
   <property name="toolTip">
    <string>输出&quot;hello world&quot;</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="action_Output2">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/output2.svg</normaloff>:/images/output2.svg</iconset>
   </property>
   <property name="text">
    <string>输出2</string>
   </property>
   <property name="toolTip">
    <string>输出&quot;你好，磐优&quot;</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="action_Output3">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/output3.svg</normaloff>:/images/output3.svg</iconset>
   </property>
   <property name="text">
    <string>输出3</string>
   </property>
   <property name="toolTip">
    <string>输出&quot;uDDS.cn&quot;</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="action_Output4">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/output4.svg</normaloff>:/images/output4.svg</iconset>
   </property>
   <property name="text">
    <string>输出4</string>
   </property>
   <property name="toolTip">
    <string>输出&quot;很高兴参加本次面试&quot;</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="action_Output5">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/output5.svg</normaloff>:/images/output5.svg</iconset>
   </property>
   <property name="text">
    <string>输出5</string>
   </property>
   <property name="toolTip">
    <string>输出&quot;I Love C++/QT&quot;</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyWidget</class>
   <extends>QWidget</extends>
   <header location="global">mywidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections/>
</ui>
